scanner/scanner.go

```diff
@@ -122,6 +123,48 @@ func (s *Scanner) isNewLineChar(c rune) bool {
 	return false
 }
 
+func (s *Scanner) isWhiteChar(c rune) bool {
+	return c == ' ' || c == '\t'
+}
+
+func (s *Scanner) isOnlyWhiteToLineEnds(src []rune, size, idx int) bool {
```

Comment:
@goccy 

Thank you for commenting.

Before begin, in line 126, maybe `isWhiteSpaceChar` is more clear.
- [From YAML spec](https://yaml.org/spec/1.2.2/#55-white-space-characters), blank space + tab characters combined symbol is called `White Space Characters`.
- This function aims to detect whether a single char is included in the White Space Characters or not.
- Also, `isNewLineChar` is already exists. Following `isXxxxChar` is better.

And then, How about `isWhiteSpaceCharUntilNewLineChar` on this function?
- Just combined: isWhiteSpaceChar until NewLineChar?
- BTW, [from YAML spec](https://yaml.org/spec/1.2.2/#54-line-break-characters), it says `Line Break Characters`, but there is already exists `isNewLineChar`. So NewLine is more clear instead BreakLine.


I can provide some more ideas with these combinations:

1. is/are
2. WhiteSpaceChar/WhiteSpaceChars
3. To/Until
4. NewLine/LineBreak/LineEnds

Reply:
I think this name ( `isWhiteSpaceCharUntilNewLineChar` or `isOnlyWhiteToLineEnds` ) itself is good. What I am wondering is that with this English meaning, the following string would be `true`. But in fact, it is `false`.

```
<white-space><white-space><new-line>
```

Specifically, I don't understand why it is returning false when isWhiteChar 🤔 .

Reply:
I'm sorry. I miss remembered the details actually this PR was created 2 months ago, and I found there are some mistakes such as `for i := idx + 1; idx < size; idx++` (`i` is not incrementing) :bow:

---

YAML spec says...

> All leading and trailing white space characters on each line are excluded from the content.

ref https://yaml.org/spec/1.2.2/#example-double-quoted-line-breaks

There is already covered "leading" part.

```
		} else if s.isWhiteChar(c) && isFirstLineChar {
			continue
```
ref. https://github.com/goccy/go-yaml/pull/342/files#diff-7d1df5e49f35d124e1d892865498cde1749af35afbb0f0022499a4b1b09a91eaR365-R366

This function aims to cover "trailing" part.

---

Now, I have another idea for this topic:

- Create a new process to support "trailing" part instead this `isOnlyWhiteToLineEnds` discussion.
- In `scanDoubleQuote` function, if whitespace is found, try to scan the next whitespaces until another char.
    - If there are multiple whitespaces, keep them as the buffer.
    - If another char is newline, drop those whitespaces and continue to scanDoubleQuote loop.
    - If another char is not newline, the buffer is joined to `value` to handle as whitespace chars and continue to scanDoubleQuote loop.
    - If reached the size limit, same as "another char is not newline" case. (btw, this is the unexpected case because it should have `"` as end of double quote)

The reason for why "scan" instead "check", it avoids multiple times check whitespaces like this:

```
aaa<whites><whites><whites><newline>bbb
```

---

What do you think?

Reply:
coding image:
```go
} else if s.isWhiteChar(c) {
    // scan multiple whitespaces
    buf := []rune{}
    for ; idx < size; idx++ {
        c := src[idx]
        if !s.isWhiteChar(c) {
            break
        }
        buf = append(buf, c)
    }

    // skip if whitespaces are in leading or trailing
    if isFirstLineChar || s.isNewLineChar(src[idx]) {
        idx-- // handle latest char in main scan loop.
        continue
    }

    // handle as whitespace chars if intermediate whitespaces
    value = append(value, buf...)
    idx-- // handle latest char in main scan loop.
    continue
```